apps/server/src/modules/tool/external-tool/uc/external-tool.uc.spec.ts

```
@@ -273,6 +274,159 @@ describe(ExternalToolUc.name, () => {
+					})
+				);
+				expect(externalToolService.createExternalTool).toHaveBeenNthCalledWith(
+					2,
+					new ExternalTool({
+						...externalTool2.getProps(),
+						id: expect.any(String),
+					})
+				);
+			});
+
+			it('should return a result report', async () => {
+				const { user, externalTool1, externalTool2 } = setup();
+
+				const result = await uc.importExternalTools(user.id, [externalTool1.getProps(), externalTool2.getProps()]);
+
+				expect(result).toEqual<ExternalToolImportResult[]>([
+					{
+						toolName: externalTool1.name,
+						mediumId: externalTool1.medium?.mediumId,
+						mediumSourceId: externalTool1.medium?.mediaSourceId,
+						toolId: externalTool1.id,
+					},
+					{
+						toolName: externalTool2.name,
+						mediumId: externalTool2.medium?.mediumId,
+						mediumSourceId: externalTool2.medium?.mediaSourceId,
+						toolId: externalTool2.id,
+					},
+				]);
+			});
+		});
+
+		describe('when an external tools fails the validation', () => {
+			const setup = () => {
+				const user = userFactory.buildWithId();
+				const externalTool1 = externalToolFactory.build({
+					name: 'tool1',
+					medium: {
+						mediumId: 'medium1',
+						mediaSourceId: 'mediumSource1',
+					},
+				});
+				const externalTool2 = externalToolFactory.build({
+					name: 'tool1',
+					medium: {
+						mediumId: 'medium2',
+						mediaSourceId: 'mediumSource2',
+					},
+				});
+				const error = new Error('same tool name');
+
+				authorizationService.getUserWithPermissions.mockResolvedValue(user);
+				logoService.fetchLogo.mockResolvedValueOnce(undefined);
+				toolValidationService.validateCreate.mockResolvedValueOnce();
+				externalToolService.createExternalTool.mockResolvedValueOnce(externalTool1);
+				toolValidationService.validateCreate.mockRejectedValueOnce(error);
+
+				return {
+					user,
+					externalTool1,
+					externalTool2,
+					error,
+				};
+			};
+
+			it('should check the users permission', async () => {
```

should return an error in the result report (or something along these lines ^^)